### PR TITLE
Add a readme in examples directory

### PR DESCRIPTION
**Pull request by awly**
Explain that examples are specific to TPM device versions.
Rename genaik to tpm-genaik to match the rest.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# go-tpm usage examples
+
+This directory contains binaries demonstrating usage of go-tpm.
```
Comment (tmroeder) [inline examples/README.md]: [optional nit]: This sentence could be simplified and made clearer at the same time:

"This directory contains binaries that show how to use go-tpm."

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+## Versions
+
+All directories that start with `tpm-` are for TPM 1.x devices.
```
Comment (tmroeder) [inline examples/README.md]: [optional]: another simplification: drop the "All" from each sentence:

"Directories that start with "...

File: examples/README.md (hunk 1 above)
Comment (awly) [reply examples/README.md]: Done

File: examples/README.md (hunk 2 above)
Comment (awly) [reply examples/README.md]: Done

**Pull request merged**